curl-helper.c

```diff
@@ -5377,6 +5377,115 @@ value caml_curl_check_enums(value v_unit)
   CAMLreturn(v_r);
 }
 
+#if LIBCURL_VERSION_NUM >= 0x075600
```

Comment:
Orthogonal to the patch, but:

Up until now we detected features in `libcurl` at compile-time by using a discovery script (see `discover.ml`). But perhaps it would be simpler to do as in your patch and simply use version guards everywhere to conditionalize compilation (instead of the current `HAVE_` macros). It would allow us to get rid of the discovery script, I think.

To be discussed.